enterprise/backend/src/metabase_enterprise/query_field_validation/api.clj

```diff
@@ -25,6 +25,30 @@
                                       [:= :qf.field_id :f.id]
                                       [:= :f.active false]]])
 
+(defn- inactive-fields [cards]
+  (when (seq cards)
+    (t2/select :model/QueryField
+               {:select [:qf.card_id
+                         [:f.name :field]
+                         [:t.name :table]
+                         [:t.active :table_active]]
+                :from   [[(t2/table-name :model/QueryField) :qf]]
+                :join   [[(t2/table-name :model/Field) :f] [:= :f.id :qf.field_id]
+                         [(t2/table-name :model/Table) :t] [:= :t.id :f.table_id]]
+                :where  [:and
+                         [:= :qf.explicit_reference true]
+                         [:= :f.active false]
+                         [:in :card_id (map :id cards)]]})))
+
+(defn- inactive-errors [inactive-fields]
+  (reduce
+   (fn [acc {:keys [field table table_active]}]
+     (update acc (if table_active :inactive-fields :inactive-tables) conj
+             {:table table, :field field}))
+   {:inactive-fields []
+    :inactive_tables []}
```

Comment:
that level of consistency between dashes and underscores scares me tbh 😁

Reply:
Why not group-by though? `(group-by #(if (:table_active %) :inactive-fields :inactive-tables) inactive-fields)` and you're done

Reply:
The inconsistency is totally accidental, yeah that would be crazy.

I thought about using group by, but then it'll leave that extra key in the values. I could use another pass to strip it, but that feels more complex.

Better group-by would do the trick 😊

Reply:
We've decided to flatten this actually, so no group-by required.